phpseclib/Crypt/Random.php

```diff
@@ -148,13 +148,13 @@ function crypt_random_string($length)
             session_start();
 
             $v = $seed = $_SESSION['seed'] = pack('H*', sha1(
-                serialize($_SERVER) .
-                serialize($_POST) .
-                serialize($_GET) .
-                serialize($_COOKIE) .
-                serialize($GLOBALS) .
-                serialize($_SESSION) .
-                serialize($_OLD_SESSION)
+                phpseclib_safe_serialize($_SERVER) .
+                phpseclib_safe_serialize($_POST) .
+                phpseclib_safe_serialize($_GET) .
+                phpseclib_safe_serialize($_COOKIE) .
+                phpseclib_safe_serialize($GLOBALS) .
+                phpseclib_safe_serialize($_SESSION) .
+                phpseclib_safe_serialize($_OLD_SESSION)
```

Comment:
> I guess we could still do isset() but idk... issues arising from disabled functions are not issues that we should be overly concerned with imho. You disable functions at your own peril.

True, but disabling `session_start()` seems to be a valid configuration option. (Unlike, e.g., a patch to the PHP interpreter itself.)

To the other Superglobals:
* `$_OLD_SESSION`: Does this one exist? It's not mentioned in https://secure.php.net/manual/en/language.variables.superglobals.php
* `$GLOBALS`: This one seems to actually always exist.

* Others: It seems that they may be non-existent, too. From https://secure.php.net/manual/en/language.variables.superglobals.php:
  > Note: Variable availability
  > By default, all of the superglobals are available but there are directives that affect this availability. For further information, refer to the documentation for [variables_order](https://php.net/manual/en/ini.core.php#ini.variables-order). 

Those are just some thoughts anyway.